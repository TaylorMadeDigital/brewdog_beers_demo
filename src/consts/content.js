export const CONTENT = {
  global: {
    title: 'Brewdog Beer Collection',
    placeholder: '- Select an option -',
    logo: 'Brewdog logo',
  },
  listing: {
    brewed: 'First brewed:',
    abv: 'ABV:',
    ibu: 'IBU:',
  },
  detail: {
    brewersTips: 'Brewer\'s Tip',
    foodPairing: 'Food Pairing',
    ingredients: 'Ingredients',
    notAvailable: 'Not supplied',
    footer: 'Contributed by -',
  },
  filter: {
    title: 'Order by:',
  }
};
