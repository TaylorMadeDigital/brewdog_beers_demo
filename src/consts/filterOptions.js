export const FILTER_OPTIONS = [
    {
      value: 'name',
      label: 'Name'
    },
    {
      value: 'first_brewed',
      label: 'Date first brewed'
    },
    {
      value: 'abv',
      label: 'ABV'
    },
    {
      value: 'ibu',
      label: 'IBU'
    },
  ]