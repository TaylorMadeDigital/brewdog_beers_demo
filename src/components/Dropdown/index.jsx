import React from 'react';
import PropTypes from 'prop-types';
import { CONTENT } from '../../consts/content';

const Dropdown = ({
  id,
  value,
  onChange,
  placeholder,
  options
}) => {
  return (
      <select
        id={id}
        data-testid="dropdown"
        value={value}
        onChange={e => onChange(e.target.value)}>
          <option>{placeholder}</option>
        {options.map((o, i) => (
          <option key={i} value={o.value}>{o.label}</option>
        ))}
      </select>
  );
};

export const dropdownPropTypes = {
  id: PropTypes.string,
  value: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.string,
    label: PropTypes.string,
  })).isRequired,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

Dropdown.propTypes = dropdownPropTypes;

Dropdown.defaultProps = {
  placeholder: CONTENT.global.placeholder,
};

export default Dropdown;
