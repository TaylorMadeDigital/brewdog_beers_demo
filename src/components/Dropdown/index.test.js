import React from 'react';
import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import Dropdown from './index';
import { FILTER_OPTIONS } from '../../consts/filterOptions';

describe('<Dropdown />', () => {

  afterEach(cleanup);

  it('snapshot: renders default state', () => {
    const { asFragment } = render(<Dropdown value={FILTER_OPTIONS[0].label} onChange={() => {}} options={FILTER_OPTIONS} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should select the right option', () => {
    const mockMyEventHandler = jest.fn()
    render(<Dropdown value={FILTER_OPTIONS[1].label} onChange={mockMyEventHandler}  options={FILTER_OPTIONS} />);
    fireEvent.change(screen.getByTestId("dropdown"), {
      target: { value: "name" },
    })
    expect(mockMyEventHandler).toHaveBeenCalledWith('name')
  })
});
