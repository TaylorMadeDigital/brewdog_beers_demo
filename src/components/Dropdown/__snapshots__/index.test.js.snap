// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> snapshot: renders default state 1`] = `
<DocumentFragment>
  <select
    data-testid="dropdown"
  >
    <option>
      - Select an option -
    </option>
    <option
      value="name"
    >
      Name
    </option>
    <option
      value="first_brewed"
    >
      Date first brewed
    </option>
    <option
      value="abv"
    >
      ABV
    </option>
    <option
      value="ibu"
    >
      IBU
    </option>
  </select>
</DocumentFragment>
`;
